### PR TITLE
Added training Scheduler files

### PR DESCRIPTION
**Pull request by Shalika-Madhushanki**

### CODE REVIEW

File: src/trainer/JasmineGraphTrainingSchedular.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            vertexCountToPartitionId.insert({vertexcount + centralVertexCount, partition_id});
+
+
+        }
+
+        float totMemEstimation = 0;
+        int iteration = 0;
+        for (auto i = vertexCountToPartitionId.begin(); i != vertexCountToPartitionId.end(); ++i) {
+            totMemEstimation += estimateMemory(i->first);
+            if (totMemEstimation < 600000000) {  //total=65843492
```
Comment (miyurud) [inline src/trainer/JasmineGraphTrainingSchedular.cpp]: We should get numbers such as 600000000 defined as constants in a header file. Also what is the rationale for selecting 600000000 as the value here?

File: src/trainer/JasmineGraphTrainingSchedular.cpp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+
+}
+
+float JasmineGraphTrainingSchedular::estimateMemory(int vertexCount) {
+    float featureMetrixSize;
+    float adjacencyMetrixSize;
+    float degreeMetrixSize;
+    float embeddingMetrixSize;
+    int featureCount = 100;
```
Comment (miyurud) [inline src/trainer/JasmineGraphTrainingSchedular.cpp]: Are we using 100 features every round of training? Can this be taken as a function argument?

File: src/trainer/JasmineGraphTrainingSchedular.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    float networkXgraphsize;
+
+    featureMetrixSize = 4 * 16 * featureCount * (vertexCount + 1);
```
Comment (miyurud) [inline src/trainer/JasmineGraphTrainingSchedular.cpp]: Should the variable name be changed to "featureMatrixSize"? I think you are referring a matrix here. If this is the case please update the below variables as well with similar name change.

File: src/trainer/JasmineGraphTrainingSchedular.cpp (hunk 1 above)
Comment (Shalika-Madhushanki) [reply src/trainer/JasmineGraphTrainingSchedular.cpp]: This value is taken by chinthaka ayya's work. Dinka has implemented it. she'll replace it here.

File: src/trainer/JasmineGraphTrainingSchedular.cpp (hunk 2 above)
Comment (Shalika-Madhushanki) [reply src/trainer/JasmineGraphTrainingSchedular.cpp]: feature count will be taken from the database. Dinika will add it once this is merged

File: src/trainer/JasmineGraphTrainingSchedular.cpp (hunk 3 above)
Comment (Shalika-Madhushanki) [reply src/trainer/JasmineGraphTrainingSchedular.cpp]: Yes. I'll change it

File: src/trainer/JasmineGraphTrainingSchedular.cpp (hunk 1 above)
Comment (miyurud) [reply src/trainer/JasmineGraphTrainingSchedular.cpp]: Ok

File: src/trainer/JasmineGraphTrainingSchedular.cpp (hunk 2 above)
Comment (miyurud) [reply src/trainer/JasmineGraphTrainingSchedular.cpp]: Ok

File: src/trainer/JasmineGraphTrainingSchedular.cpp (hunk 3 above)
Comment (miyurud) [reply src/trainer/JasmineGraphTrainingSchedular.cpp]: Ok, this has been corrected now

**Pull request merged**